calculator.go

```diff
@@ -1,13 +1,91 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
-// Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+import (
+	"errors"
+	"math"
+)
+
+var (
+	// errNotEnoughNumbers is returned by functions that expects
+	// minimum of two arguments.
+	errNotEnoughNumbers = errors.New("minimum two numbers are required")
+
+	errDivisionByZero     = errors.New("division by zero")
+	errSqrtNumberNegative = errors.New("sqrt from negative number")
+)
+
+// Add takes 2 or more numbers and returns the result of adding them together.
+// If the number of arguments is less than two the function will return an error.
+func Add(a ...float64) (float64, error) {
+	if len(a) < 2 {
+		return 0, errNotEnoughNumbers
+	}
+	var result float64
+
+	for _, i := range a {
+		result = result + i
```

Comment:
There's a nice `+=` operator in Go, which means we can write this more concisely.